modules/install-ibm-cloud-configuring-the-public-subnet.adoc

```diff
@@ -0,0 +1,152 @@
+
+.Procedure
+
+. Install `dnsmasq`:
++
+[source,terminal]
+----
+$ sudo dnf install dnsmasq
+----
+
+. Set an unused IP address from the public IP range:
++
+[source,terminal]
+----
+$ UNUSEDIP=<IP-address>
+----
+
+. Open the `dnsmasq` configuration file:
++
+[source,terminal]
+----
+$ sudo vi /etc/dnsmasq.conf
+----
+
+. Add the following configuration to the `dnsmasq` configuration file:
++
+[source,configuration files]
+----
+interface=baremetal
+except-interface=lo
+bind-dynamic
+log-dhcp
+
+dhcp-range=$UNUSEDIP,$UNUSEDIP,$PUBLICCIDR
+dhcp-option=baremetal,121,0.0.0.0/0,$BAREMETAL_GATEWAY,$BOOTSTRAP_PRIV_IP,$BSHOSTIP
+
+dhcp-hostsfile=/var/lib/dnsmasq/dnsmasq.hostsfile
+----
++
+[NOTE]
+====
+The Dnsmasq configuration uses the BASH variables created when preparing the provisioner node. If you install the DHCP server on a node other than the provisioner node, you will need to re-create these BASH variables.
+====
+
+. Obtain the list of hardware for the cluster using the IBM Cloud command line interface:
++
+[source,terminal]
+----
+$ ibmcloud sl hardware list
+----
+
+. Obtain the MAC addresses and IP addresses for each node using the IBM Cloud command line interface:
++
+[source,terminal]
+----
+$ ibmcloud sl hardware detail <ID> --output JSON | jq '.networkComponents[] | "\(.primaryIpAddress) \(.macAddress)"' | grep -v null
+----
++
+Replace `<ID>` with the ID of the node.
++
+.Example output
+[source,terminal]
+----
+"10.196.130.144 00:e0:ed:6a:ca:b4"
+"141.125.65.215 00:e0:ed:6a:ca:b5"
+----
++
+Make a note of the MAC address and IP address of the public network. Make a separate note of the MAC address of the private network, which you will use later in the `install-config.yaml` file. Repeat this procedure for each node until you have all the public MAC and IP addresses for the public `baremetal` network, and the MAC addresses of the private `provisioning` network.
+
+. Add the MAC and IP address pair of the public `baremetal` network for each node into the `dnsmasq.hostsfile` file:
++
+[source,terminal]
+----
+$ sudo vim /var/lib/dnsmasq/dnsmasq.hostsfile
+----
++
+.Example input
+[source,configuration files]
+----
+00:e0:ed:6a:ca:b5,141.125.65.215,master-0
+<MAC>,<IP>,master-1
+<MAC>,<IP>,master-2
+<MAC>,<IP>,worker-0
+<MAC>,<IP>,worker-1
+...
+----
++
+Replace `<MAC>,<IP>` with the public MAC address and public IP address of the corresponding node name.
+
+. Check the status of the `dnsmasq` service:
++
+[source,terminal]
+----
+$ systemctl status dnsmasq
+----
+
+. If `dnsmasq` is running, restart it:
++
+[source,terminal]
+----
+$ sudo systemctl restart dnsmasq
+----
+
+. If `dnsmasq` isn't running, start it:
++
+[source,terminal]
+----
+$ sudo systemctl start dnsmasq
+----
+
+. Enable `dnsmasq` so that it starts when booting the node:
++
+[source,terminal]
+----
+$ sudo systemctl enable dnsmasq
+----
+
```

Comment:
No need to check the status, or start. These can be removed.
You can just do 
sudo systemctl restart dnsmasq
sudo systemctl enable dnsmasq